resolwe/flow/executors/run.py

```diff
@@ -116,88 +94,77 @@ def get_stdout(self):
+        """
+        packet = {
+            ExecutorProtocol.DATA_ID: DATA['id'],
+            ExecutorProtocol.COMMAND: cmd,
+        }
+        packet.update(extra_fields)
+
+        # TODO what happens here if the push fails? we don't have any realistic recourse,
+        # so just let it explode and stop processing
+        queue_channel = SETTINGS['REDIS_CHANNEL_PAIR'][0]
         try:
-            # Ensure that we only update the fields that were changed.
-            data.save(update_fields=update_fields)
-
-            if kwargs.get('status', None) == Data.STATUS_ERROR:
-                self.process_failed = True
-                logger.error(
-                    __("Error occured while running a '{}' process.", self.process.name),
-                    extra={
-                        'data_id': self.data_id,
-                        'api_url': '{}{}'.format(
-                            getattr(settings, 'RESOLWE_HOST_URL', ''),
-                            reverse('resolwe-api:data-detail', kwargs={'pk': self.data_id})
-                        ),
-                    }
-                )
-
-        except ValidationError as exc:
-            data = Data.objects.get(pk=self.data_id)
-
-            data.process_error.append(exc.message)
-            data.status = Data.STATUS_ERROR
-            self.process_failed = True
-
-            logger.error(
-                __("Error occured while running a '{}' process.", self.process.name),
-                exc_info=True,
-                extra={
-                    'data_id': self.data_id,
-                    'api_url': '{}{}'.format(
-                        getattr(settings, 'RESOLWE_HOST_URL', ''),
-                        reverse('resolwe-api:data-detail', kwargs={'pk': self.data_id})
-                    ),
-                }
-            )
-
-            update_fields = ['process_error', 'status']
-
-            try:
-                data.save(update_fields=update_fields)
-            except:  # pylint: disable=bare-except
-                pass
+            self.redis.rpush(queue_channel, json.dumps(packet))
+        except:  # pylint: disable=bare-except
+            logger.error("Error sending command to manager:\n\n%s", traceback.format_exc())
+            raise
+
+        if not expect_reply:
+            return
+
+        while True:
+            for _ in range(3):
+                response = self.redis.blpop(self.queue_response_channel, timeout=1)
+                if response:
+                    break
+            else:
+                # NOTE: If there's still no response after three seconds, the system is broken
+                # enough that it makes sense to give up; we're isolated here, so if the manager
+                # doesn't respond, we can't really do much more than just crash
+                raise RuntimeError("No response from the manager after 3 retries.")
+
+            _, item = response
+            packet = json.loads(item.decode('utf-8'))
```

Comment:
I don't think anything sensible can be done to "handle" errors here, other than log them and abort. Also, they shouldn't happen at all anyway, since JSON serialization obviously succeeded on the other side of the wire.